pymc_marketing/customer_choice/mnl_logit.py

```diff
@@ -0,0 +1,586 @@
+        | Depvar     | alt_1_X1   | alt_1_X2   | alt_2_X1   | alt_2_X2   |
+        +============+============+============+============+============+
+        | alt_1      | 2.4        | 4.5        | 5.4        | 6.7        |
+        +------------+------------+------------+------------+------------+
+        | alt_2      | 3.5        | 6.7        | 2.3        | 8.9        |
+        +------------+------------+------------+------------+------------+
+
+    Example `utility_equations` list:
+
+    >>> utility_equations = [
+    ...     'alt_1 ~ X1_alt1 + X2_alt1 | income',
+    ...     'alt_2 ~ X1_alt2 + X2_alt2 | income',
+    ...     'alt_3 ~ X1_alt3 + X2_alt3 | income'
+    ... ]
+
+    """
+
+    _model_type = "Multinomial Logit Model"
+    version = "0.1.0"
+
+    def __init__(
+        self,
+        choice_df: pd.DataFrame,
+        utility_equations: list[str],
+        depvar: str,
+        covariates: list[str],
+        model_config: dict | None = None,
+        sampler_config: dict | None = None,
+    ):
+        self.choice_df = choice_df
+        self.utility_equations = utility_equations
+        self.depvar = depvar
+        self.covariates = covariates
+
+        model_config = model_config or {}
+        model_config = parse_model_config(model_config)
+
+        super().__init__(model_config=model_config, sampler_config=sampler_config)
+
+    @property
+    def default_model_config(self) -> dict:
+        """Default model configuration.
+
+        This is a Categorical likelihood, Normal intercept,
+        and a Normal vector of beta coefficients
+
+        Returns
+        -------
+        dict
+            The default model configuration.
+
+        """
+        alphas = Prior("Normal", mu=0, sigma=5, dims="alts")
+        betas = Prior("Normal", mu=0, sigma=1, dims="alt_covariates")
+        betas_fixed = Prior("Normal", mu=0, sigma=1, dims=("alts", "fixed_covariates"))
+
+        return {
+            "alphas_": alphas,
+            "betas": betas,
+            "betas_fixed_": betas_fixed,
+            "likelihood": Prior(
+                "Categorical",
+                p=0,
+                dims="obs",
+            ),
+        }
+
+    @property
+    def default_sampler_config(self) -> dict:
+        """Default sampler configuration."""
+        return {}
+
+    @property
+    def output_var(self) -> str:
+        """The output variable of the model."""
+        return "y"
+
+    @property
+    def _serializable_model_config(self) -> dict[str, int | float | dict]:  # type: ignore
+        result: dict[str, int | float | dict] = {
+            "alphas_": self.model_config["alphas_"].to_dict(),
+            "likelihood": self.model_config["likelihood"].to_dict(),
+            "betas": self.model_config["betas"].to_dict(),
+            "betas_fixed": self.model_config["betas_fixed_"].to_dict(),
+        }
+
+        return result
+
+    def parse_formula(self, df, formula, depvar):
```

Comment:
Can we pull out this into a function instead of method. A good separation might be: 
1. parse_formula(formula) -> tuple[target, alt_covariates, fixed_covariates]
2. check_columns(covariates, df)
3. check_dependent_variable(target, ser)

Reply:
Yeah, can do... probably not tonight though. Wrecked. Will try and pick it up during the week.